checkstyle.xml

```diff
@@ -0,0 +1,38 @@
+<?xml version="1.0" encoding="utf-8" ?>
+
+<!DOCTYPE module PUBLIC
+        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
+        "https://checkstyle.org/dtds/configuration_1_3.dtd">
+
+<module name="Checker">
+    <module name="TreeWalker">
+        <module name="ConstantName"/>
+
+        <!-- Validate that first sentence is not empty and first sentence is not missing -->
+        <module name="SummaryJavadocCheck"/>
+        <module name="InvalidJavadocPosition"/>
+
+        <module name="JavadocType">
+            <property name="scope" value="package"/>
+        </module>
+        <module name="MissingJavadocType">
+            <property name="scope" value="package"/>
+        </module>
```

Comment:
Why are these package scope but not public?

Reply:
> Using a specific scope means not only will that modifier be examined, but also all the modifiers listed above it in the previous list. Specifying public means only items with public modifiers are checked. Specifying protected means only public and protected modifiers are checked. If you wish to only validate items with private modifiers and ignore any others, then you must set the exclude scope property, if available, to the scope above it in the table. In this case you would exclude package.

Reference: https://checkstyle.sourceforge.io/property_types.html#Scope

Reply:
Would it help if I add a comment about this in the `checkstyle` configuration?

Reply:
Ah, right, I did not know the scopes worked like that, but it makes sense.

That's not necessary IMO, it's a core part of how Checkstyle works. I just don't use Checkstyle very often and so wasn't aware.